src/wms/src/main/java/org/geoserver/wms/WMSInfoImpl.java

```diff
@@ -349,4 +351,13 @@ public GrowableInternationalString getInternationalRootLayerAbstract() {
     public void setInternationalRootLayerAbstract(InternationalString rootLayerAbstract) {
         this.internationalRootLayerAbstract = InternationalStringUtils.growable(rootLayerAbstract);
     }
+
+    @Override
+    public List<String> getAllowedURLsForAuthForwarding() {
+        return allowedURLsForAuthForwarding;
```

Comment:
You can make check that the field is null, and set it to a new ArrayList if that's the case.
(not thread-safe, but config is not supposed to be modified concurrently... if you want belts and suspenders, also make the method synchronized or do a double-checked locking with volatile field).

Reply:
Ok. I simply did as a simple null check since I have found other pieces of code doing the same check)

Reply:
So if the collection is null, the only way to make it non-null is to cast to the Impl or use reflection to access the setter. It's not ideal... other bits of code might be doing it, but it's not really how it should be done....

Reply:
My fault. During a refactoring of another branch and another repo on IntelliJ I made some mistakes with git stash & co.
When I said "I simply did a null check" I was meaning that I have simply adopted the null check + new ArrayList as in other pieces of code instead of using the synchronized / volatile approach. 
However, for some reason that code change didn't end in the commit (due to the above IntelliJ mess) so I have re-updated it right now.

Sorry for the confusion :D